version: "3.2"

services:
  adminer:
    container_name: adminer
    image: adminer:latest
    restart: always
    ports:
      - 8080:8080
    networks:
      - app-net
  league:
    container_name: league
    build:
      context: .
      dockerfile: ./build/Dockerfile
    restart: on-failure
    volumes:
      - ./migrations:/home/app/migrations
      - ./bin:/home/app/bin
      - ./tests:/home/app/tests
      - ./seeds:/home/app/seeds
      - ./src:/home/app/src
      - ./statics:/home/app/statics
      - ./manage.py:/home/app/manage.py
      - ./setup.py:/home/app/setup.py
    env_file:
      - ${HUNCHO_DIR}/env/dev/.env.league
    healthcheck:
      test: curl --fail -s league:5000/ping || exit 1
      interval: 60s
      timeout: 5s
      retries: 5
    ports:
      - 20010:5000
    depends_on:
      - league_db
    networks:
      - app-net
  league_db:
    container_name: league_db
    image: postgres:9.6
    volumes:
      - ${HUNCHO_DIR}/data/dev/league_db/data:/var/lib/postgresql/data/
    env_file:
      - ${HUNCHO_DIR}/env/dev/.env.league_db
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U league" ]
      interval: 10s
      timeout: 5s
      retries: 20
    networks:
      - app-net
networks:
  app-net:
    driver: bridge